### PR TITLE
pyspread: fix build and simplify expression

### LINKED ISSUE #117896: Pyspread fails to build
Pyspread fails to build with the following log:
```
< many successful and irrelevant lines >
pyspread> adding 'pyspread-1.99.5.dist-info/LICENSE'
pyspread> adding 'pyspread-1.99.5.dist-info/METADATA'
pyspread> adding 'pyspread-1.99.5.dist-info/WHEEL'
pyspread> adding 'pyspread-1.99.5.dist-info/top_level.txt'
pyspread> adding 'pyspread-1.99.5.dist-info/RECORD'
pyspread> removing build/bdist.linux-x86_64/wheel
pyspread> Finished executing setuptoolsBuildPhase
pyspread> glibPreInstallPhase
pyspread> glibPreInstallPhase
pyspread> installing
pyspread> Executing pipInstallPhase
pyspread> /build/pyspread-1.99.5/dist /build/pyspread-1.99.5
pyspread> Processing ./pyspread-1.99.5-py3-none-any.whl
pyspread> Installing collected packages: pyspread
pyspread> Successfully installed pyspread-1.99.5
pyspread> /build/pyspread-1.99.5
pyspread> install: missing destination file operand after '/nix/store/dqga5kh5jvcli4prkh02nzyxxnb608cs-pyspread-1.99.5/share/applications'
pyspread> Try 'install --help' for more information.
```
 To me, it seems like the postInstall hook of [pyspread/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/pyspread/default.nix) is broken:
```
  postInstall = ''
    runHook preInstall
    install -D $out/share/applications
    install -m 644 $desktopItem/share/applications/* $out/share/applications
    runHook postInstall
  '';
```
Specifically, the second line should not be a valid use of `install`? I have no idea why this hasn't been a problem earlier.

Pyspread packager: @AndersonTorres

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
Closes https://github.com/NixOS/nixpkgs/issues/117896

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @lokegustafsson

### CODE REVIEW

**Comment by lokegustafsson** [comment]
Thanks!

File: pkgs/applications/office/pyspread/default.nix [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qtsvg
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
```
Comment (AndersonTorres) [inline pkgs/applications/office/pyspread/default.nix:36]: Can someone explain me what is the rationale/standard here? 

Is it to use `python3.pkgs` or `python3Packages`?

It is to declare these packages explicitly at the input parameters, or to expose them via `with`?

File: pkgs/applications/office/pyspread/default.nix [hunk 2]
```diff
@@ -50,20 +54,11 @@ buildPythonApplication rec {
   };
 
   postInstall = ''
-    runHook preInstall
-    install -D $out/share/applications
-    install -m 644 $desktopItem/share/applications/* $out/share/applications
-    runHook postInstall
+    install -m 644 -Dt $out/share/applications ${desktopItem}/share/applications/*
```
Comment (AndersonTorres) [inline pkgs/applications/office/pyspread/default.nix:57]: As far as I remember, there is not an explicit installer.

File: pkgs/applications/office/pyspread/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/office/pyspread/default.nix:36]: > Is it to use `python3.pkgs` or `python3Packages`?

This only matters if you also input python(3) which could get mixed python versions when used together with python3Packages.

> It is to declare these packages explicitly at the input parameters, or to expose them via with?

If you do not declare them explicitly they cannot be easily overwritten.

Comment (dotlambda) [reply pkgs/applications/office/pyspread/default.nix:36]: > If you do not declare them explicitly they cannot be easily overwritten.

It's the exact opposite: You should override `python3` to override a python package. So `python3` should be the only argument.

**Comment by AndersonTorres** [comment]
Can someone test Darwin?

File: pkgs/applications/office/pyspread/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/office/pyspread/default.nix:36]: > > If you do not declare them explicitly they cannot be easily overwritten.
> 
> It's the exact opposite: You should override `python3` to override a python package. So `python3` should be the only argument.

But Python is not what the programs wait, but Python libraries.

Comment (dotlambda) [reply pkgs/applications/office/pyspread/default.nix:36]: That's what I mean with python package. If you want to override `python3.pkgs.xyz`, you need to use `python3.override { packageOverrides = ... }`.

File: pkgs/applications/office/pyspread/default.nix [hunk 3]
```diff
@@ -1,44 +1,50 @@
...
 
-  src = fetchPypi {
+  src = python3.pkgs.fetchPypi {
     inherit pname version;
     sha256 = "sha256-05bC+Uvx72FAh3qxkgXm8jdb/gHRv1D/M7tjOEdE3Xg=";
   };
 
-  pythonLibs = [
+  patches = [
+    # https://gitlab.com/pyspread/pyspread/-/merge_requests/34
```
Comment (AndersonTorres) [inline pkgs/applications/office/pyspread/default.nix:19]: Is this expected to be included next release?

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/applications/office/pyspread/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/applications/office/pyspread/default.nix:19]: I hope so.

**Comment by risicle** [comment]
> Can someone test Darwin?

WFM macos 10.14 (though most of the icons are missing, perhaps this was always the case on macos)

**Pull request merged**